tests/testthat/test_plots.R

```diff
@@ -3,95 +3,173 @@ context("Basic time series graphs")
-    expect_is(plot(tm, smoother = FALSE), "patchwork")
-    expect_is(suppressWarnings(plot(tm, compare = FALSE)), "gtable")
-    expect_is(
-        suppressWarnings(plot(tm, compare = FALSE, multiplicative = TRUE)),
-        "gtable"
-    )
-    expect_is(
-        suppressWarnings(plot(tm, smoother = FALSE, compare = FALSE)),
-        "gtable"
-    )
+    expect_is(plot(tm, var_name), "patchwork")
+    expect_is(plot(tm, var_name, mult_fit = TRUE), "patchwork")
+    expect_is(plot(tm, var_name, smoother = FALSE), "patchwork")
+    expect_false(ggplotable(plot(tm, var_name)))
 })
 
-test_that("Unsupported plots error", {
-    expect_warning(decompositionplot(t))
-    expect_warning(recompose(decompositionplot(t)))
-    expect_warning(decompositionplot(tm))
-    expect_warning(recompose(decompositionplot(tm)))
-    expect_error(seasonplot(tm))
-    expect_warning(capture.output(forecastplot(tm)))
+test_that("Annual data forecast is fine", {
+    t <- inzightts(visitorsA2)
+    tm <- inzightts(visitorsA2, var = 2:5)
+    expect_is(plot(predict(t)), "ggplot")
+    expect_is(plot(predict(tm)), "ggplot")
 })
 
-test_that("Annual data", {
-    t <- iNZightTS(visitorsA2)
-    tm <- iNZightTS(visitorsA2, var = 2:5)
-    expect_is(plot(t), "ggplot")
-    expect_warning(
-        plot(t, forecast = 2),
-        "Forecasting not available for annual data"
-    )
-    expect_is(plot(tm), "ggplot")
-    expect_is(plot(tm, compare = FALSE), "gtable")
-})
+t_neg <- t
+t_neg$Australia[1] <- -1
+y <- visitorsQ %>%
+    dplyr::mutate(key = rep(c("A", "B", "C"), each = 18)) %>%
+    inzightts(key = "key")
 
-## clean up
-unlink("Rplot.pdf")
+test_that("Check raw-plot configuration", {
+    expect_is(plot(t, xlim = c(NA, NA)), "ggplot")
+    expect_warning(plot(t_neg, mult_fit = TRUE))
+    expect_is(plot(t, xlim = c(2005, 2010)), "ggplot")
+    # TODO: not giving a warning
+    expect_warning(plot(t, xlim = lubridate::ymd(c(20050101, 20101231))))
+    x <- inzightts(visitorsQ, var = 2:5)
+    expect_error(plot(x, c("Australia", "Japan"), ylab = "AUS"))
+    expect_warning(plot(x, names(x)[-1], aspect = 4))
+    expect_is(plot(t, aspect = 1), "ggplot")
+    expect_equal(guess_plot_var(t, "test"), c("", "test"))
+    expect_is(guess_plot_var(t, NULL), "name")
+    expect_is(plot(y), "ggplot")
+    expect_is(plot(y, compare = FALSE), "ggplot")
+    expect_is(plot(y, emphasise = 2), "ggplot")
+    expect_is(plot(y, emphasise = 2, non_emph_opacity = 1e-6), "ggplot")
+    expect_is(plot(y, emphasise = 2, non_emph_opacity = 0), "ggplot")
+    x <- visitorsM2 %>%
+        dplyr::mutate(y = ggplot2::cut_number(Japan, 4)) %>%
+        inzightts()
+    y <- visitorsA2 %>%
+        tidyr::pivot_longer(!Time, names_to = "Country", values_to = "Visitors") %>%
+        dplyr::mutate(y = ggplot2::cut_number(Visitors, 4)) %>%
+        inzightts(key = "Country")
+    expect_is(plot(x, "y"), "ggplot")
+    # TODO: spitting out a warning
```

Comment:
All the warnings below this line were expected and also tested via other `expect_warning()` calls. Some previous versions of `testthat` allowed warnings to exist when testing with `expect_is()` thus I did not suppress them.